### PR TITLE
Add deprecation call policy

### PR DESCRIPTION
**Pull request by ManifoldFR**
This PR adds a shortcut call policy for marking an exposed function as deprecated.

### CODE REVIEW

**Comment by ManifoldFR** [comment]
damn I forgot to add a changelog again

**Comment by ManifoldFR** [comment]
I've modified the PR as discussed @jcarpent

**Review by jcarpent — DISMISSED** [review]

File: include/eigenpy/deprecation-policy.hpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+template <DeprecationType deprecation_type = DeprecationType::DEPRECATION,
+          class BasePolicy = bp::default_call_policies>
+struct deprecated_function
+    : deprecation_warning_policy<deprecation_type, BasePolicy> {
+  static constexpr char defaultMsg[] =
+      "This function has been marked as deprecated, and will be "
+      "removed in the future.";
+
+  deprecated_function<>(const std::string &msg = defaultMsg)
```
Comment (jcarpent) [inline include/eigenpy/deprecation-policy.hpp:63]:
```suggestion
  deprecated_function(const std::string &msg = defaultMsg)
```

**Comment by ManifoldFR** [comment]
My last commit seems to have fixed the (linker) issue, so we should be able to merge now

**Review by jcarpent — APPROVED** [review]

**Pull request merged**